### PR TITLE
estimate area under a curve defined by non-negative real-valued continuous function within a continuous interval using monte-carlo

### PR DESCRIPTION
**Pull request by singlav**
estimates area under a curve defined by non-negative real-valued continuous function within a continuous interval using monte-carlo

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/monte_carlo.py [hunk 1]
```diff
@@ -35,34 +36,43 @@ def in_circle(x: float, y: float) -> bool:
     print("The total error is ", abs(pi - pi_estimate))
 
 
-def area_under_line_estimator(iterations: int,
+def area_under_curve_estimator(iterations: int,
+                              function_to_integrate: Callable[[float], float],
```
Comment (cclauss) [inline maths/monte_carlo.py]: `Callable[[float], float]` ?  I would have thought `Callable[float, float]`.  What does the extra bracket do for us?

File: maths/monte_carlo.py [hunk 2]
```diff
@@ -35,34 +36,43 @@ def in_circle(x: float, y: float) -> bool:
...
-        b. Expected value = average of those values
-    4. Actual value = (max_value^2 - min_value^2) / 2
+        b. Evaluate f(x) at each of the drawn x values
+        c. Expected value = average of the function evaluations
+    4. Estimated value of integral = Expected value * (max_value - min_value)
     5. Returns estimated value
     """
-    return mean(uniform(min_value, max_value) for _ in range(iterations)) * (max_value - min_value)
+    
+    return mean(function_to_integrate(uniform(min_value, max_value)) for _ in range(iterations)) * (max_value - min_value)
```
Comment (cclauss) [inline maths/monte_carlo.py]: Please run your code through [black](https://github.com/psf/black) to auto-fix long lines and do other formatting.  All text (comments included) should be wrapped at 88 characters max.

File: maths/monte_carlo.py [hunk 3]
```diff
@@ -35,34 +36,43 @@ def in_circle(x: float, y: float) -> bool:
...
 
 
 def area_under_line_estimator_check(iterations: int,
                                     min_value: float=0.0,
                                     max_value: float=1.0) -> None:
     """
-    Checks estimation error for area_under_line_estimator func
-    1. Calls "area_under_line_estimator" function
+    Checks estimation error for area_under_curve_estimator function
+    for f(x) = x where x lies in 0 to 1
```
Comment (cclauss) [inline maths/monte_carlo.py]: 0 and 1?

File: maths/monte_carlo.py [hunk 4]
```diff
@@ -35,34 +36,43 @@ def in_circle(x: float, y: float) -> bool:
...
+    1. Calls "area_under_curve_estimator" function
     2. Compares with the expected value
     3. Prints estimated, expected and error value
     """
     
-    estimated_value = area_under_line_estimator(iterations, min_value, max_value)
+    def identity_function(x: float) -> float:
```
Comment (cclauss) [inline maths/monte_carlo.py]: Now we finally have a function that needs doctests.  ;-)

File: maths/monte_carlo.py [hunk 5]
```diff
@@ -72,6 +82,29 @@ def area_under_line_estimator_check(iterations: int,
...
+    return
+
+
+def pi_estimator_using_area_under_curve(iterations: int) -> None:
+    """
+    Area under curve y = sqrt(4 - x^2) where x lies in 0 to 2
+    is equal to pi
+    """
+
+    def function_to_integrate(x: float) -> float:
```
Comment (cclauss) [inline maths/monte_carlo.py]: This function needs doctests.

File: maths/monte_carlo.py (hunk 1 above)
Comment (singlav) [reply maths/monte_carlo.py]: Thanks. I tried removing bracket. It gives following error. I think it expects a list.

TypeError: Callable[args, result]: args must be a list. Got <class 'float'>

File: maths/monte_carlo.py (hunk 5 above)
Comment (singlav) [reply maths/monte_carlo.py]: Thanks for the suggestion. I have added. I am new to doctests. Please let me know if I need to update it.

File: maths/monte_carlo.py (hunk 4 above)
Comment (singlav) [reply maths/monte_carlo.py]: Thanks. I have added it.

File: maths/monte_carlo.py (hunk 3 above)
Comment (singlav) [reply maths/monte_carlo.py]: Thanks for pointing out. I have corrected.

File: maths/monte_carlo.py (hunk 2 above)
Comment (singlav) [reply maths/monte_carlo.py]: Thanks. I ran it this time.

**Comment by cclauss** [comment]
I just learned that doctests can not find inner functions... Let’s leave them in place as pure documentation.

https://stackoverflow.com/questions/29307730/python3-doctest-helper-internal-functions

**Comment by singlav** [comment]
Thanks for all the help. :)
@TheSuperNoob FYI and thanks for the suggestion earlier.

**Pull request merged**